### ISSUE TITLE
ion-back-button defaultHref does not support relative URLs

### ISSUE THREAD

**Issue by adamduren** [ionic-team/ionic#17409]
# Bug Report

**Ionic version:**

[x] **4.0.1**

**Current behavior:**

Ionic navigates to a 404 when using `ion-back-button` with a `[defaultHref]` of a relative URL such as `../`. 

**Expected behavior:**

The [Angular Navigation Docs](https://ionicframework.com/docs/navigation/angular/#navigating-to-different-routes) state:
> A note on navigation with relative URLs: Currently, to support multiple navigation stacks, relative URLs are something not supported

Ideally until we figure out how to support multiple stacks, which may not be possible, we should maybe think about adding support via a config option for apps with a single navigation stack.

At a minimum the [ion-back-button Docs](https://ionicframework.com/docs/api/back-button) should be updated to make this more clear. I thought I had read though the Angular Navigation docs beforehand but was only made aware of the limitation by searching through the Ionic Slack history.

**Steps to reproduce:**

**Related code:**

```
N/A
```

**Other information:**

**Ionic info:** 

```
Ionic:

   ionic (Ionic CLI)             : 4.10.1 (/Users/adamduren/.nvm/versions/node/v10.14.1/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.0.1
   @angular-devkit/build-angular : 0.12.2
   @angular-devkit/schematics    : 7.2.2
   @angular/cli                  : 7.2.2
   @ionic/angular-toolkit        : 1.2.2

Cordova:

   cordova (Cordova CLI) : 8.1.2 (cordova-lib@8.1.1)
   Cordova Platforms     : android 7.1.1, browser 5.0.4, ios 4.5.5
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 2.2.0, (and 18 other plugins)

System:

   Android SDK Tools : 26.1.1 (/Users/adamduren/Library/Android/sdk)
   NodeJS            : v10.14.1 (/Users/adamduren/.nvm/versions/node/v10.14.1/bin/node)
   npm               : 6.6.0
   OS                : macOS Mojave
   Xcode             : Xcode 10.1 Build version 10B61
```

**Comment by tommyarn** [comment]
@adamduren I am also having a similar issue , when i use ion-back-button with [defaultHref] i get 404 error .  did you find a path forward yet ?

**Comment by adamduren** [comment]
I came up with a solution that works but it's not ideal. Here's a [gist](https://gist.github.com/adamduren/3f8e4fe5259cd7c009656497f5329b94). Hope this helps.

Note: It doesn't have to be used as a decorator. I was just wanting a reason to play around with them. It could just be a function that's made available to the template via other means. One idea is a pipe, it would need to be impure since it would rely on `window.location` global state.

**Comment by adamduren** [comment]
You could also tie into the router to get the latest url and avoid checking `window.location`.

**Comment by liamdebeasi** [comment]
Thanks for the issue. I am going to close this as this issue was resolved in Ionic Framework v4.2.0. If you are still running into issues after updating to the latest version of Ionic Framework, please open a new issue. Thanks!
